Jenkinsfile

```diff
@@ -24,64 +24,84 @@ pipeline {
         }
       }
       steps {
-        error("A maintainer needs to approve this PR for CI by commenting")
+        error('A maintainer needs to approve this PR for CI by commenting')
       }
     }
 
-    stage('Test Suite') {
-      matrix {
-        axes {
-          axis {
-              name 'GO_VERSION'
-              values '1.14', '1.16'
-          }
+    stage('Test') {
+      agent {
+        node {
+          label 'ec2-fleet'
+          customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}"
         }
+      }
 
-        agent {
-          node {
-            label 'ec2-fleet'
-            customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}-${GO_VERSION}"
+      steps {
+        script {
+          withCredentials([
+            string(credentialsId: 'logdna-gpg-key', variable: 'GPG_KEY'),
```

Comment:
Did you have to add this with infra?

Reply:
if you add `keyFileVariable: 'some_variable_name'` to the `withCredentials` parameters, then Jenkins will put the credentials in a temporary location on the worker, and store the location in the `'some_variable_name'` environment variable. that is probably safer than the echo you're doing a few lines down (although +1 for deleting it in an `always` block).

although if the location is stored in the variable, i'm not sure off the top of my head how you'd inject that in the Dockerfile

Reply:
this already existed and matches what the CDN is using